BatteryStateManagement_proj.c

```diff
@@ -0,0 +1,209 @@
+  {
+   printf("Batter parameter %s -->  %s !\n",batPar[BattParmt_str_p->batParIndex],batLevel[BattParmt_str_p->batParIndex]);
+   return 0;
+  }
+  else
+  {
+   printf("Batter parameter %s -->  %s !\n",batPar[BattParmt_str_p->batParIndex],batLevel[BattParmt_str_p->batParIndex]);
+   return 1;
+  }
+}
+
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryWarnHandling_v
+ */
+/*!    \brief       predecting  the warning for max range
+ * 
+ *     \param       NA 
+ *     \returns     NA
+ *
+*//*------------------------------------------------------------------------*/
+void batteryWarnHandling_v()
+{
+ static float range;
+ /*warning level percentage calulcator : 5% of max range*/
+  BattParmt_str_p->warnLevel = (0.05 * BattParmt_str_p->maxRange);
+  range = (BattParmt_str_p->maxRange - BattParmt_str_p->warnLevel);
+ 
+   /*warning prediction max - Approaching charge-peak*/
+    if ((BattParmt_str_p->batteryParameter >= range) && (BattParmt_str_p->batteryParameter < BattParmt_str_p->maxRange) )
+    {
+     printf("Warn---> %s---->Entering Max\n",batPar[BattParmt_str_p->batParIndex]);
+    }
+    else
+    {
+     batteryWarnHandlingMinRange_v();
+    }
+}
+
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryWarnHandlingMinRange_v
+ */
+/*!    \brief       predecting  the warning for min range
+ * 
+ *     \param       NA 
+ *     \returns     NA
+ *
+*//*------------------------------------------------------------------------*/
+void batteryWarnHandlingMinRange_v()
+{
+ static float range;
+ /*warning level percentage calulcator : 5% of max range*/
+ 
+  range = (BattParmt_str_p->minRange +  BattParmt_str_p->warnLevel);
+ 
+   /*warning prediction min - Approaching discharge*/
+    if ( (BattParmt_str_p->batteryParameter > BattParmt_str_p->minRange) &&  (BattParmt_str_p->batteryParameter <= range))
+    {
+     printf("Warn---> %s---->Entering Min\n",batPar[BattParmt_str_p->batParIndex]);
+    }
+}
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryStateValidation_i
+ */
+/*!    \brief       Monitoring the battery states
+ * 
+ *     \param       temperature,soc,chargeRate
+ *     \returns     batterystate -OK/NOK
+ *
+*//*------------------------------------------------------------------------*/
+int batteryStateValidation_i(float temperature, float soc, float chargeRate)
+{ 
+ 
+ int retTempStat_i, retsocStat_i,retchargeStat_i;
+ 
+  if ( GERMAN_LANG_SLECTD != langSelected_uint)
+  {
+  /*Validation done with current battery parameter,Min and max range in english lang*/
+   
+  BattParmt_str_p->batteryParameter =temperature;
+  BattParmt_str_p->minRange = 0;
+  BattParmt_str_p->maxRange =45;
+  BattParmt_str_p->batParIndex =0;
+  batteryWarnHandling_v();
+  retTempStat_i   = batteryCondMonitor_i();
```

Comment:
Pass the above structure (using a const pointer) to `batteryCondMonitor_i` for thread-safety.
This will also make it a pure function.